src/libaktualizr/primary/aktualizr_test.cc

```diff
@@ -895,6 +895,53 @@ TEST(Aktualizr, InstallWithUpdates) {
   }
 }
 
+/**
+ * Verifies reporting of update download progress
+ *
+ * Checks actions:
+ *
+ * - [x] Report download progress
+ */
+TEST(Aktualizr, ReportDownloadProgress) {
+  // The test initialization part is repeated in many tests so maybe it makes sense
+  // to define a fixture and move this common initialization part into the fixture's SetUp() method
+  TemporaryDirectory temp_dir;
+  auto http = std::make_shared<HttpFake>(temp_dir.Path(), "hasupdates");
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, http->tls_server);
+  auto storage = INvStorage::newStorage(conf.storage);
+  Aktualizr aktualizr(conf, storage, http);
+
+  /* perhaps should be static const member of DownloadProgressReport */
+  const unsigned int download_complete_progress_value = {100};
+  unsigned int download_progress = {0};
+  std::function<void(std::shared_ptr<event::BaseEvent> event)> report_event_hdlr =
+      [&](const std::shared_ptr<event::BaseEvent>& event) {
+        /* perhaps should be static const member of DownloadProgressReport */
+        if (event->variant != "DownloadProgressReport") {
+          return;
+        }
+
+        auto download_progress_event = std::dynamic_pointer_cast<event::DownloadProgressReport>(event);
+        ASSERT_NE(download_progress_event, nullptr);
+        ASSERT_NE(download_progress_event.get(), nullptr);
+        EXPECT_GT(download_progress_event->progress, download_progress);
```

Comment:
Should probably be `EXPECT_GE` in case the percentage hasn't actually increased.

Reply:
ok